lib/src/autoimpl.rs

```diff
@@ -303,7 +303,7 @@ impl ImplTraits {
             Ok(Item::Enum(item)) => self.expand_enum(item, find_impl),
             Ok(Item::Struct(item)) => self.expand_struct(item, find_impl),
             Ok(item) => {
-                emit_error!(item, "expected struct");
+                item.span().error("expected struct").emit_as_expr_tokens();
                 Toks::new()
```

Comment:
Like here

Reply:
Oh okey I was thinking that this was emitting looking at the other crates that made this migration. Sorry I'm not very familiar with details of proc-macro. Will try to drop-in replacement with the crate you mentioned.

Reply:
I get the feeling that other people may have made this mistake.

I started working on this including error-verification a while back, but gave up: https://github.com/kas-gui/impl-tools/commits/work/
Might as well just do the switch to `proc-macro-error2` now though.

Reply:
Yeah I think a lot of proc-macro crate lack of error tests, I think mainly because they are boring/complex/.. to do, maybe with new IA tools it could be improved.